pkg/api/runner.go

```diff
@@ -63,7 +65,15 @@ func withSemaphore(fns []Request, maxConcurrency int) Request {
 				defer func() {
 					<-sem
 				}()
-				errCh <- fn(ctx, ch)
+
+				if requestTimeout == 0 {
+					errCh <- fn(ctx, ch)
```

Comment:
Is there any way of making it so we only have one place where we write to the channel? Having multiple branches is harder to debug / maintain over time.

Reply:
we have to pass different contexts for the timeout vs no-timeout case, assuming preserving no-timeout is desired. If this were Python I could use a ternary statement to pass the context, but it's not 😛. Would it be better or worse to declare var reqErr above, set reqErr in each branch, and then write that to the channel? I suppose another option would be to set cctx to either a new context.WithTimeout or the original parent context, do the branching there, and only have one execution of fn

Reply:
Something like the following would probably work (assuming you keep `time.Duration` vs `*time.Duration`:
```
if requestTimeout > 0 {
    var cancel context.CancelFunc
    ctx, cancel = context.WithTimeout(ctx, requestTimeout)
    defer cancel()
}
errCh <- fn(ctx, ch)
```